pyro/distributions/sine_skewed.py

```diff
@@ -0,0 +1,92 @@
+from math import pi
+
+import torch
+from torch.distributions import Uniform
+
+from pyro.distributions import constraints
+
+from .torch_distribution import TorchDistribution
+
+
+class SineSkewed(TorchDistribution):
+    """The Sine Skewed distribution [1] is a distribution for breaking pointwise-symmetry on a base-distribution over
+    the d-dimensional torus.
+
+    This distribution requires the base distribution on a torus. The parameter skewness can be inferred using
+    :class:`~pyro.infer.HMC` or :class:`~pyro.infer.NUTS`. For example, the following will produce a uniform prior
+    over skewness for the 1-torus,::
+
+        def model(...):
+            ...
+            skewness_phi = pyro.sample(f'skewness_phi', Uniform(skewness.abs().sum(), 1 - tots))
+            psi_bound = 1 - skewness_phi.abs()
+            skewness_psi = pyro.sample(f'skewness_psi', Uniform(-psi_bound, psi_bound)
+            skewness = torch.stack((skewness_phi, skewness_psi), dim=0)
+            ...
+
+    .. note:: An event in the base distribution must be on a d-torus, so the event_shape must be (d, 2) or (2,).
+
+    .. note:: For the skewness parameter, it must hold that the sum of the absolute value of its weights for an event
+        must be less than or equal to one. See eq. 2.1 in [1].
+
+    ** References: **
+      1. Sine-skewed toroidal distributions and their application in protein bioinformatics
+         Ameijeiras-Alonso, J., Ley, C. (2019)
+
+    :param base_density: base density on a d-dimensional torus.
+    :param skewness: skewness of the distribution.
+    """
+    arg_constraints = {'skewness': constraints.independent(constraints.interval(-1., 1.), 1)}
+
+    support = constraints.independent(constraints.real, 1)
+
+    def __init__(self, base_density: TorchDistribution, skewness, validate_args=None):
+        assert base_density.event_shape[-1] == 2 and len(base_density.event_shape) <= 2
+        assert base_density.shape()[len(base_density.shape()) - len(skewness.shape):] == skewness.shape
```

Comment:
Remove this assertation (see the broadcasting logic in a comment below). This assertation does not support broadcasting. For example, this will fail
```
base_dist.shape = batch_shape + event_shape = () + (100, 2)
skewness.shape = (1000, 100, 1000, 2)
```

Reply:
I see; I'll take it out.